overlays/token/template.js

```diff
@@ -1,24 +1,57 @@
 const { Image } = require('canvas')
 
+const subLabelFontSize = 14;
+
 module.exports = ({
   gridsize,
   size,
   color,
   fontcolor,
   fontsize,
   label,
+  subLabel,
   image,
   font
 }, ctx) => {
-  const whitelineModifier = size < 41 ? 1.5 : 2;
-  const radius = (size + 1) / 2 - 3;
-  const xy = size < gridsize ? (gridsize + 1) / 2 : (size + 1) / 2;
+
+  let borderWidth = 3;
+  let boxEdgeWidth = 2;
+  let whitelineModifier = 3;
+  let pixelAdjustment = 0.5;
+  if (size <= 60) {
+    borderWidth = 1;
+    boxEdgeWidth = 1;
+    whitelineModifier = 1.5;
+  } else if (size <= 120) {
+    borderWidth = 2;
+    boxEdgeWidth = 1;
+    whitelineModifier = 2;
+    pixelAdjustment = 0;
+  } 
+
+  const snapToPx = (num) => {
+    return Math.floor(num) + pixelAdjustment;
+  }
+
+  const snapToSinglePx = (num) => {
+    return Math.floor(num) + 0.5;
+  }
+
+  const radius = snapToPx((size + 1) / 2 - 3);
+  const xy = Math.floor(size < gridsize ? (gridsize + 1) / 2 : (size + 1) / 2);
   const imageTL = size < gridsize ? (gridsize - size) / 2 : 0;
   let tokenEdgeColour = '#07031a';
+  let hasSubLabel = image && size >= 40 && subLabel;
 
   // inner token edge
   ctx.beginPath();
```

Comment:
hasSubLabel will end up being a string not a bool here. Still truthy but maybe nicer to be explicit. Could wrap the whole thing in a Boolean() or use !!()